lib/version_utils.pm

```diff
@@ -578,9 +578,20 @@ sub get_os_release {
     my ($go_to_target, $os_release_file) = @_;
     $go_to_target    //= '';
     $os_release_file //= '/etc/os-release';
-    my %os_release = script_output("$go_to_target cat $os_release_file") =~ /^([^#]\S+)="?([^"\r\n]+)"?$/gm;
-    %os_release = map { uc($_) => $os_release{$_} } keys %os_release;
-    my ($os_version, $os_service_pack) = split(/\.|-sp/i, $os_release{VERSION});
+
+    my %os_release;
+    my $sep = (is_vmware || is_hyperv) ? "\r\n" : "\n";
+
+    foreach my $line (split(/$sep/, script_output("$go_to_target cat $os_release_file"))) {
```

Comment:
seems to me that the only change that you need is 
```
my $sep = (is_vmware || is_hyperv) ? "\r\n" : "\n";                                                                                                                                                                                      
    my %os_release = script_output("$go_to_target cat $os_release_file") =~ /^([^#]\S+)="?([^"$sep]+)"?$/gm;
```
if i am not wrong. Why do you need a loop? in any case, in the way that is implemented now, i would suggest to move the `script_output` outside the loop and keep the `dial` . wdyt?

Reply:
I find Martin's solution more readable. The previous solution is looping over the hash as well.